### DISCUSSION TITLE
I want open a dialog after mainWindow, but get error "Cannot show window with non-visible owner."

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#13163] [Q&A] [answered]
Then I got after 
`InitializeComponent();`
the mainWindow is still loading.

When the mainWindow finishes the process, where can I put the code to open a dialog?

**Comment by thevortexcloud** [comment]
Are you just trying to show a dialogue as soon as the main window opens?

**Reply by IronEaterBeast** [reply]
Yes, I put the code just behind
`InitializeComponent();`

I got an error.

**Reply by thevortexcloud** [reply, marked as answer]
The window does not really "exist" at that point. To be completely safe you should instead trigger it on the `Loaded` event. At that point the window should be in a state where you can actually do things with it. That said, it will also be in an even weirder state if you do anything before `InitializeComponent` is called since that is what actually causes the XAML file to be loaded.

**Reply by IronEaterBeast** [reply]
I think you are right, but I can't find any information about `Loaded` in the Avalonia document (https://docs.avaloniaui.net/docs/next/welcome). Is it Version: 11.0.0?

**Reply by thevortexcloud** [reply]
It looks like it is indeed in >=11 only. It's not a Window specific event. It comes from [`Control.cs`](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Control.cs#L184C31-L188) and as such all controls have it. Instead it looks like you want either the `Opened` event (which is a Window specific event) or the `AttachedToVisualTree` event in .10.
